pkg/kubernetes/status/readiness.go

```diff
@@ -0,0 +1,95 @@
+package status
+
+import (
+	"context"
+	"fmt"
+
+	kudoapi "github.com/kudobuilder/kudo/pkg/apis/kudo/v1beta1"
+	label "github.com/kudobuilder/kudo/pkg/util/kudo"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// ComputeReadiness updates instance Status.Conditions.Ready based on current state of underlying resources
+// currently readiness examines the following types: Pods, StatefulSets, Deployments, ReplicaSets and DaemonSets
+// Instance is considered ready if all the resources linked to this instance are also ready
+func IsReady(i kudoapi.Instance, c client.Client) (ready bool, msg string, err error) {
+	resources, err := healthResources(c, i.Name, i.Namespace)
+	if err != nil {
+		return false, "", err
+	}
+	ready = true
+	readinessMessage := ""
+	for _, res := range resources {
+		healthy, _, err := IsHealthy(res)
+		if !healthy {
+			readinessMessage += fmt.Sprintf(", %v", err)
+			ready = false
+		}
+	}
+
+	return ready, readinessMessage, nil
+}
+
+func healthResources(c client.Client, instanceName, instanceNamespace string) ([]runtime.Object, error) {
+	instanceLabels, err := labels.Parse(fmt.Sprintf("%s=%s,%s=%s", label.InstanceLabel, instanceName, label.HeritageLabel, "kudo"))
+	if err != nil {
+		return nil, fmt.Errorf("unable to create list of labels to define health: %v", err)
+	}
+
+	dList := &appsv1.DeploymentList{}
+	err = c.List(context.TODO(), dList, &client.ListOptions{Namespace: instanceNamespace, LabelSelector: instanceLabels})
```

Comment:
Couldn't we just list _every_ resource owned by the instance, then run `IsHealthy` for each of them? `IsHealthy` will consider unknown resources as healthy.

Reply:
no, the request always has to contain type, so there's no way to query multiple types in one request

Reply:
That's unfortunate :(
We will then have to update this list if we add new resources to `status.IsHealthy`. Can you add a comment in `pkg/kubernetes/status/health.go` that this list needs to be updated if items are added there?

Reply:
that's actually not true, not all items we know how to compute health for need to be considered for readiness. Readiness is currently just "runnable" resources like deployments etc. E.g. not even jobs are considered for example

Reply:
You're right, I didn't consider this. Got it now!